pkg/cmd/secret/set/set.go

```diff
@@ -0,0 +1,216 @@
+package set
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghinstance"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/secret/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/nacl/box"
+)
+
+type SetOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	RandomOverride io.Reader
+
+	SecretName      string
+	OrgName         string
+	Body            string
+	Visibility      string
+	RepositoryNames []string
+}
+
+func NewCmdSet(f *cmdutil.Factory, runF func(*SetOptions) error) *cobra.Command {
+	opts := &SetOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "set <secret name>",
```

Comment:
Minor: we did not call this out explicitly in https://github.com/cli/cli/blob/trunk/docs/command-line-syntax.md, but I think that placeholder names would look better without spaces, e.g. `<secret-name>`. What do you think?

My reasoning is that because shells typically do word-splitting on spaces, it's prudent to avoid a space when demonstrating a single argument.